dags/openshift_nightlies/scripts/install/rosa.sh

```diff
@@ -137,6 +137,83 @@ _adm_logic_check(){
     echo "Failed to execute oc adm commands after 100 attempts with 5 sec interval" 
 }
 
+_balance_infra(){
+    echo "Initiate migration of prometheus componenets to infra nodepools"
+    oc get pods -n openshift-monitoring -o wide | grep prometheus-k8s
+    oc get sts prometheus-k8s -n openshift-monitoring
+    echo "Restart stateful set pods"
+    oc rollout restart -n openshift-monitoring statefulset/prometheus-k8s 
+    echo "Wait till they are completely restarted"
+    oc rollout status -n openshift-monitoring statefulset/prometheus-k8s
+    echo "Check pods status again and the hosting nodes"
+    oc get pods -n openshift-monitoring -o wide | grep prometheus-k8s
+    for node in $(oc get pods -n openshift-monitoring -o wide | grep -i prometheus-k8s | grep -i running | awk '{print$7}');
+    do
+        if [[ $(oc get nodes | grep infra | awk '{print$1}' | grep $node) != "" ]]; then
+                echo "$node is an infra node"
+        else
+                echo "ERROR: Prometheus-k8s pod on $node is not an infra node"
```

Comment:
Lets look to add in a retry of the rebalance in the case that it fails (which I've seen on some occasions).